tests/09-setuidgid.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+### Constants
+c_valgrind_min=1
+test_stderr="${s_basename}-stderr.txt"
+
+# Keep this in sync with the value in tests/setuidgid/main.c!
+EX_NOPERM=77
+
+test_setuidgid_username_groupname() {
+	username_groupname=$1
+	pass_exitcode=$2
+
+	setup_check_variables
+
+	printf "=== check ${s_count}\n" >> ${test_stderr}
+	cd ${scriptdir}/setuidgid/ && \
+	    ${c_valgrind_cmd} ./test_setuidgid "${username_groupname}"	\
+	    2>> ${test_stderr}
+	cmd_exitcode=$?
+
+	if [ ${cmd_exitcode} -eq "$EX_NOPERM" ]; then
+		# Skip this check, since we can only really test setuidgid
+		# as root (or otherwise have permission to change the username
+		# and group list).
+		echo "-1" > ${c_exitfile}
```

Comment:
Hmm, is this the right test here?  Shouldn't we skip if we're non-root?  I mean, right now we wouldn't treat it as an error if `test_setuidgid` *always* returned EX_NOPERM...?

Reply:
Fair enough.  There's only 1 test that doesn't give `EX_NOPERM`, but I've change it so that it makes sure it doesn't get that for that test.